render_pdf/src/parser.py

```diff
@@ -0,0 +1,130 @@
+from lxml import etree
+from lxml.etree import fromstring
+import json
+from collections import defaultdict
+from urllib.request import urlopen
+
+def get_body(root):
+    return root.find(tagname("text")).find(tagname("body"))
+
+def get_title(root, alt_root):
+    # God this is ugly # TODO
+    try:
+        return root.find(tagname("head")).text
+    except:
+        try:
+            root = alt_root
+            tag = root.find(tagname("teiHeader"))\
+                    .find(tagname("fileDesc")) \
+                    .find(tagname("titleStmt"))
+            return tag.find(tagname("title")).text
+        except:
+            print("[Warning] Couldn't get title.")
+            return ""
```

Comment:
semantics seem tighter if returning `None` then adding `-> Optional[str]` to the function signature